### PR TITLE
Respect user-selected file format when saving

### PR DESCRIPTION
**Pull request by dg0yt**
This PR adds missing pieces, cleans up the API, and applies to ocd-export.

### CODE REVIEW

File: src/gui/main_window.cpp [hunk 1]
```diff
@@ -1132,7 +1143,7 @@ bool MainWindow::showSaveAsDialog()
 	// Fails when using different formats for import and export:
 	//	Q_ASSERT(FileFormats.findFormatForFilename(path) == format);
 	
-	return savePath(path);
+	return savePath(path, format);
```
Comment (dg0yt) [inline src/gui/main_window.cpp]: Hint: This is the root of the changes: We have got the format from the file dialog's *filter* and must pass it down.

File: src/fileformats/ocd_file_format.cpp [hunk 2]
```diff
@@ -45,6 +72,16 @@ OcdFileFormat::OcdFileFormat()
 	// Nothing
 }
 
+OcdFileFormat::OcdFileFormat(quint16 version)
+: FileFormat { MapFile, idForVersion(version),
```
Comment (lpechacek) [inline src/fileformats/ocd_file_format.cpp]: Both these entries and "OCAD" from default constructor (line 69) are presented in the Save as... dialog as possible file types. Only code inspection reveals that "OCAD" means "save with v9 or the version from which the map was loaded". A more descriptive text is needed.

File: src/fileformats/ocd_file_format.cpp [hunk 3]
```diff
@@ -45,6 +72,16 @@ OcdFileFormat::OcdFileFormat()
...
+               version == 1 ? ::OpenOrienteering::ImportExport::tr("OCAD legacy export")
```
Comment (lpechacek) [inline src/fileformats/ocd_file_format.cpp]: Legacy export should mention the version number. E.g. "OCAD legacy exporter (v8)". Not every user knows...

File: src/fileformats/ocd_file_format.cpp (hunk 2 above)
Comment (dg0yt) [reply src/fileformats/ocd_file_format.cpp]: Proposal: "OCAD (auto-selected version)"

Details might go to a manual page on file formats (or OCD format support).

File: src/fileformats/ocd_file_format.cpp (hunk 3 above)
Comment (dg0yt) [reply src/fileformats/ocd_file_format.cpp]: Proposal: "OCAD version 8 (old exporter)". 

Anyway, we should remove the legacy exporter as soon as possible. On Android, we don't have this dialog at all.

**Comment by dg0yt** [comment]
I will merge all but the OCD-related commit directly to `dev` so that it becomes available for testing and for PR #1068.

**Comment by lpechacek** [comment]
The proposals look good to me. Go ahead!

**Comment by dg0yt** [comment]
OCD-related commit will go to ocd-export branch.

**Pull request merged**